extensions/mapstore/src/main/java/com/hazelcast/mapstore/FromSqlRowConverter.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.mapstore;
+
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.nio.serialization.genericrecord.GenericRecord;
+import com.hazelcast.nio.serialization.genericrecord.GenericRecordBuilder;
+import com.hazelcast.sql.SqlColumnMetadata;
+import com.hazelcast.sql.SqlRow;
+import com.hazelcast.sql.SqlRowMetadata;
+
+import javax.annotation.Nonnull;
+
+final class FromSqlRowConverter {
+
+    private FromSqlRowConverter() {
+    }
+
+
+    // Convert SqlRow to GenericRecord
+    @Nonnull
+    public static GenericRecord toGenericRecord(SqlRow sqlRow, GenericMapStoreProperties properties) {
+        GenericRecordBuilder builder = GenericRecordBuilder.compact(properties.compactTypeName);
```

Comment:
I think GenericRecordBuilder is a more general concept. It looks independent from SQL. I don't think adding a method would be appropriate

Reply:
Eventually, I would like to have the connectors to produce a row with a single column containing compact record. This would avoid the overhead of converting native object from SQL connector (jdbc result, or in mongo the bson document) -> SqlRow -> compact.

But we take one step at a time. Let's keep this as is now.